src/record/single-notifier.ts

```diff
@@ -107,29 +106,35 @@ export class SingleNotifier {
 
     // todo we can clean this up and do cb = (error, data) => error ? reject(error) : resolve()
     for (let i = 0; i < responses.length; i++) {
-      const response = responses[i]
-      if (response.callback) {
-        response.callback(error, data)
-      } else if (error && response.reject) {
-        response.reject(error)
-      } else if (response.resolve) {
-        response.resolve(data)
-      }
+      responses[i](error, data)
     }
     this.requests.delete(name)
     return
   }
 
   private onConnectionLost (): void {
     this.requests.forEach(responses => {
-      responses.forEach(response => {
-        if (response.callback) {
-          response.callback(EVENT.CLIENT_OFFLINE)
-        } else if (response.reject) {
-          response.reject(EVENT.CLIENT_OFFLINE)
-        }
-      })
+      responses.forEach(response => response(EVENT.CLIENT_OFFLINE))
     })
     this.requests.clear()
   }
+
+  private onExitLimbo (): void {
+    for (let i = 0; i < this.limboQueue.length; i++) {
+      const message = this.limboQueue[i]
+      const requests = this.requests.get(message.name as string)
+      if (requests) {
+        requests.forEach(cb => cb(EVENT.CLIENT_OFFLINE))
+      }
+      this.requests.delete(message.name as string)
```

Comment:
what else would be left in requests? Client is offline and all requests should be 'offlined'